main.py

```diff
@@ -29,6 +45,452 @@
+
+        # Rescale the players
+        # for player in self.players:
+        #     player.scale_player(width_scale, height_scale)
+
+        # Update screen size
+        self.screen = pygame.display.set_mode(new_resolution)
+        self.current_screen_size = new_resolution
+
+    def handle_events(self) -> None:
+        """ get events like keypress or mouse clicks """
+        for event in pygame.event.get():
+            match event.type:
+                case pygame.QUIT:
+                    pygame.quit()
+                    sys.exit()
+            
+            if event.type == pygame.MOUSEBUTTONDOWN:
+                if self.state == 'main_menu':
+                    PLAY_BUTTON, OPT_BUTTON, QUIT_BUTTON = self.main_menu()
+                    if PLAY_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.state = 'play'
+                    elif OPT_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.state = 'options'
+                    elif QUIT_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        pygame.quit()
+                        sys.exit()
+
+                elif self.state == 'play':
+                    BACK_BUTTON = self.play(tmx_maps=self.tmx_map["map"], player_start_pos="Fort")
+                    if BACK_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.state = 'main_menu'
+
+                elif self.state == 'options':
+                    BACK_BUTTON, RES_BUTTON, SFX_BUTTON, DIFF_BUTTON = self.options() #SFX_BUTTON is going to be implemented later
+                    if BACK_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.state = 'main_menu'
+                    elif RES_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.state = 'res'
+                    elif DIFF_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.state = 'diff'
+
+                elif self.state == 'diff':
+                    BACK_BUTTON = self.diff()
+                    if BACK_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.state = 'options'
+
+                # TODO: I had no time to correctly set up the resolution change, I will fix it later
+                elif self.state == 'res':
+                    BACK_BUTTON, FHD_BUTTON, HD_BUTTON, THIRD_BUTTON = self.res()
+                    if BACK_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.state = 'options'
+                    if FHD_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.scale_content((1920, 1080))
+                    if HD_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.scale_content((1280, 720))
+                    if THIRD_BUTTON.checkForInput(pygame.mouse.get_pos()):
+                        self.scale_content((800, 600))
+
+        if self.state == 'main_menu':
```

Comment:
I found a minor bug when you change the resolutions, I think it is caused because you call the tmx_maps two times because you made the GUI class again.